CRM/Odoosync/Sync/Inbound/Transaction.php

```diff
@@ -170,64 +241,125 @@ private function syncTransactions() {
-      return (int) $financialTrxn["id"];
+      $entityFinancialTrxn = civicrm_api3('EntityFinancialTrxn', 'create', [
+        'entity_table' => 'civicrm_financial_item',
+        'entity_id' => $financialItemId,
+        'financial_trxn_id' => $financialTrxnId,
+        'amount' =>  $this->validatedParams['total_amount']
+      ]);
+
+      return (int) $entityFinancialTrxn['id'];
     }
     catch (CiviCRM_API3_Exception $e) {
       $this->syncResponse['is_error'] = 1;
-      $this->syncResponse['error_message'] = ts("Can't create financial transaction");
+      $this->syncResponse['error_message'] = ts("Can't create connect 'financial item' to transaction");
       return FALSE;
     }
   }
 
   /**
-   * Creates connect transaction to contribution
+   * Gets financial item id
    *
-   * @param $financialTrxnId
-   * @param $contributionId
-   * @param $amount
+   * @return bool|int
+   */
+  public function getFinancialItemId() {
+    $query = "
+      SELECT financial_item.id AS financial_item_id
+      FROM civicrm_line_item AS line_item
+      LEFT JOIN civicrm_financial_item AS financial_item
+        ON line_item.id = financial_item.entity_id
+      WHERE line_item.contribution_id = %1 
+        AND line_item.entity_table = 'civicrm_contribution'
+        AND financial_item.entity_table = 'civicrm_line_item'
+	  ";
+    $dao = CRM_Core_DAO::executeQuery($query, [1 => [$this->validatedParams['contribution_id'], 'Integer']]);
+
+    while ($dao->fetch()) {
+      return (int) $dao->financial_item_id;
+    }
+
+    return FALSE;
+  }
+
+
+  /**
+   * Creates financial transaction
    *
-   * @return bool
+   * @return bool|int
    */
-  private function createEntityFinancialTrxn($financialTrxnId, $contributionId, $amount, $entityTable) {
+  private function createFinancialTrxn() {
     try {
-      $entityFinancialTrxn = civicrm_api3('EntityFinancialTrxn', 'create', [
-        'entity_table' => $entityTable,
-        'entity_id' => (int)$contributionId,
-        'financial_trxn_id' => (int)$financialTrxnId,
-        'amount' => $amount
+      $params = array_merge($this->validatedParams, [
+        'is_payment' => 1,
+        'net_amount' => $this->validatedParams['total_amount']
       ]);
-      return (int) $entityFinancialTrxn['id'];
+      $financialTrxn = civicrm_api3('FinancialTrxn', 'create', $params);
+
+      return (int) $financialTrxn["id"];
     }
     catch (CiviCRM_API3_Exception $e) {
-      $this->syncResponse['is_error'] = 1;
-      $this->syncResponse['error_message'] = ts("Can't create connect transaction to contribution");
       return FALSE;
     }
   }
 
+  /**
+   * Updates contribution status
+   */
+  private function updateContributionStatus() {
+    $currentContributionStatus = $this->getCurrentContributionStatus();
+
+    if ($currentContributionStatus == $this->validatedParams['contribution_status_id']) {
+      return;
+    }
+
+    $query = "
+      UPDATE civicrm_contribution AS contribution
```

Comment:
why not using the API ?

Reply:
We use SQL because CiviCRM API has limit for status changes. It may create issues with it.

Reply:
what do u mean by "limit" ?

Reply:
CiviCrm has collection of avaliable statuses for every contribution status on which it can be changed. It means that contribution status can't be changed to all existing statuses in CiviCRM. But in Odoo statuses may have other dependings (to change statuses) and that's why the issue may occur.

Reply:
You can test it in CiviCRM API explorer.